core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/ResponseRequest.java

```diff
@@ -0,0 +1,135 @@
+      // first create the history newStep
+      UpdatableHistoryStep newStep = (UpdatableHistoryStep) instanceManager.createHistoryStep();
+      newStep.setUserId(event.getUser().getUserId());
+      newStep.setAction("#response#");
+      newStep.setActionDate(event.getActionDate());
+      newStep.setUserRoleName(event.getUserRoleName());
+      if (event.getResolvedState() != null) {
+        newStep.setResolvedState(event.getResolvedState().getName());
+      }
+      // To be processed
+      newStep.setActionStatus(0);
+      newStep.setProcessInstance(processInstance);
+
+      // add the new newStep to the processInstance
+      getHistoryStepRepository().save((HistoryStepImpl) newStep);
+
+      step.set(newStep);
+      return newStep;
+    });
+
+    Transaction.performInOne(()-> {
+      UpdatableProcessInstance processInstance = getProcessInstanceRepository().getById(id);
+
+      // Do workflow stuff
+      try {
+        processEvent(processInstance, step.get().getId());
+        getProcessInstanceRepository().save((ProcessInstanceImpl) processInstance);
+      } catch (WorkflowException we) {
+        saveError(processInstance, event, step.get(), we);
+
+        throw new WorkflowException("WorkflowEngineThread.process",
+            "workflowEngine.EX_ERR_PROCESS_ADD_TASKSAVED_REQUEST", we);
+      }
+
+      return null;
+    });
+  }
+
+  /**
+   * Method declaration
+   */
+  private void processEvent(UpdatableProcessInstance instance, String stepId)
+      throws WorkflowException {
+    // get the history step
+    UpdatableHistoryStep step = (UpdatableHistoryStep) instance.getHistoryStep(stepId);
+    // only to set the current step to that step
+    instance.updateHistoryStep(step);
+
+    // add the answer
+    String answer = null;
+    try {
+      answer = (String) event.getDataRecord().getField("Content").getObjectValue();
+    } catch (FormException fe) {
+      throw new WorkflowException("WorkflowEngineThread.process", "workflowEngine.EXP_UNKNOWN_ITEM",
+          fe);
+    }
+    State state = instance.answerQuestion(answer, event.getQuestionId());
+
+    // remove the state that was discussed from the list of active states
+    instance.removeActiveState(event.getResolvedState());
+
+    // change the action status of the step
+    step.setActionStatus(1); // Processed
+    instance.updateHistoryStep(step);
+
+    // get the last participant for the discussed state
+    Participant participant = instance.getParticipant(state.getName());
+
+    // Get the task manager
+    TaskManager taskManager = WorkflowHub.getTaskManager();
+
+    // Unassign task to this participant
+    Task task = taskManager.createTask(participant, instance);
+    taskManager.unAssignTask(task);
+
+    // Remove this user of working user list
+    instance.removeWorkingUser(participant.getUser(), state, participant.getUserRoleName());
+
+    // change the action status of the step
+    // Affectations done
+    step.setActionStatus(2);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)